pkg/machines/getOSList.py

```diff
@@ -26,6 +48,14 @@
     osObj['releaseDate'] = os.get_release_date_string() or ""
     osObj['eolDate'] = os.get_eol_date_string() or ""
     osObj['codename'] = os.get_codename() or ""
+    osObj['recommendedResources'] = {}
+    recommendedResources = os.get_recommended_resources() or []
+    for resource in list(_OsinfoIter(recommendedResources)):
+        osObj['recommendedResources']['ram'] = resource.get_ram()
```

Comment:
and I don't get this. You get some list (or whatever structure) of recommended resources. And then you go through this list pick from every item ram with `get_ram` and set this value to `osObj['recommendedResources']['ram']`. So it always has the value of ram of the last item in `recommendedResources`, right?

Reply:
actually it's just one 1 object or empty array here. So the iteration is a bit confusing, I will replace it with if arr.legth > 0; arr[0]
Sorry for confusion/

Reply:
Would you prefer something like that then:

``` diff
     osObj['recommendedResources'] = {}
-    recommendedResources = os.get_recommended_resources() or []
-    for resource in list(_OsinfoIter(recommendedResources)):
-        osObj['recommendedResources']['ram'] = resource.get_ram()
+    recommendedResources = list(_OsinfoIter(os.get_recommended_resources() or []))
+    if len(recommendedResources):
+        osObj['recommendedResources']['ram'] = recommendedResources[0].get_ram()
+
     osObj['minimumResources'] = {}
-    minimumResources = os.get_minimum_resources() or []
-    for resource in list(_OsinfoIter(minimumResources)):
-        osObj['minimumResources']['ram'] = resource.get_ram()
+    minimumResources = list(_OsinfoIter(os.get_minimum_resources() or []))
+    if len(minimumResources):
+        osObj['minimumResources']['ram'] = minimumResources[0].get_ram()
 

```

Reply:
Yes, this makes much more sense :)
Do we really need `_OsinfoIter` though when there is maximum one item?
```
minimumResources = os.get_minimum_resources() # Does it return none or something else empty that can be tested on the second row?
if minimumResources:
    osObj['minimumResources']['ram'] = minimumResources.get_nth(0).get_ram()
```
would do, right? (Not sure what does `get_minimum_resources()` returns)

Reply:
Actually yes, (almost as above) - reposted.